services/traction/api/tasks/present_proof_tasks.py

```diff
@@ -0,0 +1,168 @@
+import logging
+from re import Pattern, compile
+from uuid import UUID
+from acapy_client.model.indy_proof_req_attr_spec import IndyProofReqAttrSpec
+from acapy_client.model.indy_proof_req_pred_spec import IndyProofReqPredSpec
+from acapy_client.model.indy_proof_request import IndyProofRequest
+
+from acapy_client.model.v10_presentation_send_request_request import (
+    V10PresentationSendRequestRequest,
+)
+
+
+from sqlalchemy import update
+from sqlalchemy.exc import DBAPIError
+
+
+from api.tasks.tasks import Task, TractionTaskType
+from api.db.models import Tenant
+from api.db.models.v1.contact import Contact
+from api.endpoints.models.credentials import ProofRequest
+from api.db.session import async_session
+
+from acapy_client.api.present_proof_v1_0_api import PresentProofV10Api
+from api.api_client_utils import get_api_client
+
+from api.db.models.v1.presentation_requests import VerifierPresentationRequest
+
+TRACTION_TASK_PREFIX = "traction::TASK::"
+
+
+TRACTION_SEND_PRESENT_PROOF_REQ_PATTERN = compile(
+    f"^{TRACTION_TASK_PREFIX}{TractionTaskType.send_present_proof_req}(.*)?$"
+)
+
+present_proof_api = PresentProofV10Api(api_client=get_api_client())
+
+logger = logging.getLogger(__name__)
+
+
+def get_logger(cls):
+    return logging.getLogger(cls.__name__)
+
+
+class SendPresentProofTask(Task):
+    @staticmethod
+    def _listener_pattern() -> Pattern[str]:
+        return TRACTION_SEND_PRESENT_PROOF_REQ_PATTERN
+
+    @staticmethod
+    def _event_topic() -> str:
+        return TRACTION_TASK_PREFIX + TractionTaskType.send_present_proof_req
+
+    def _get_db_model_class(self):
+        return VerifierPresentationRequest
+
+    def _get_id_from_payload(self, payload):
+        spp_t_id = payload["v_presentation_request_id"]
+        return spp_t_id
+
+    async def _perform_task(self, tenant: Tenant, payload: dict):
+        self.logger.info("> _perform_task()")
+
+        # update state from pending to 'starting'
+        # vpr =
+        # call acapy
+        contact = None
+        async with async_session() as db:
+            contact = await Contact.get_by_id(db, tenant.id, payload["contact_id"])
+            vpr = await self._get_db_model_class().get_by_id(
+                db, tenant.id, self._get_id_from_payload(payload)
+            )
+            vpr.status = "starting"
+            db.add(vpr)
+            await db.commit()
+
+        data = {
+            "body": V10PresentationSendRequestRequest(
+                connection_id=str(contact.connection_id),
+                proof_request=convert_to_IndyProofRequest(payload["proof_request"]),
+            )
+        }
+
+        resp = present_proof_api.present_proof_send_request_post(**data)
+        self.logger.warn(resp)
```

Comment:
again, let's lose the warn/warning logger statements when they are not actual warnings.